File: templates/java-webflux/subsystem/post/component/src/main/java/cobalt/post/PostRepository.java
package cobalt.post;

import cobalt.post.model.Post;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;
import java.util.stream.Collectors;

@Repository
class PostRepository {

  static List<Post> posts = List.of(
    new Post(1, 1, "Lorem ipsum dolor sit amet.", "Curabitur ultrices lacus lorem, vitae tincidunt lorem convallis."),
    new Post(2, 1, "Maecenas maximus sodales nibh vel.", "Pellentesque scelerisque tortor nec lorem volutpat, sed semper."),
    new Post(3, 3, "Sed tempus, justo at viverra.", "Cras efficitur, nunc eu maximus congue, magna quam."),
    new Post(4, 3, "Proin quis aliquam mauris. Proin.", "Donec id odio et mauris posuere auctor at."),
    new Post(5, 3, "Nulla placerat condimentum diam sed.", "In et lectus a arcu efficitur viverra in."),
    new Post(6, 3, "Etiam interdum facilisis urna, a.", "Phasellus euismod lectus eget libero viverra, sed rhoncus."),
    new Post(7, 4, "Curabitur quis leo feugiat, faucibus.", "Phasellus rutrum pretium auctor. Donec tincidunt ligula eros."),
    new Post(8, 5, "Praesent scelerisque gravida tortor a.", "Curabitur lobortis turpis at arcu facilisis, eu convallis."),
    new Post(9, 5, "Vivamus nibh nulla, aliquet quis.", "Suspendisse consequat feugiat nisi vel sollicitudin. Phasellus nec."),
    new Post(10, 5, "Aliquam elit erat, ornare nec.", "Phasellus finibus justo eget risus gravida, in venenatis."),
    new Post(11, 6, "Duis vitae placerat nibh. Integer.", "Sed blandit nisl eget nisl gravida scelerisque. Pellentesque."),
    new Post(12, 7, "Pellentesque eu tincidunt eros, sed.", "Nullam luctus dolor placerat lorem luctus, eu scelerisque."),
    new Post(13, 7, "Nulla tincidunt cursus consequat. Aliquam.", "Praesent lobortis bibendum est eu volutpat. Maecenas lobortis."),
    new Post(14, 7, "Etiam nec purus eget dui.", "Aenean interdum nec lacus eget elementum. Vivamus ultrices."),
    new Post(15, 7, "In non est fringilla eros.", "Quisque malesuada posuere mauris, a pellentesque diam aliquet."),
    new Post(16, 7, "Phasellus fermentum malesuada lectus nec.", "Nullam tortor magna, ullamcorper a elit a, aliquam.")
  );

  Flux<Post> findAll() {
    return Flux.fromIterable(posts);
  }

  Flux<Post> findByUserId(Integer userId) {
    return Flux.fromIterable(
      posts
        .stream()
        .filter(post -> post.userId.equals(userId))
        .collect(Collectors.toList())
    );
  }

  Mono<Post> findById(Integer id) {
    final Post post = posts
      .stream()
      .filter(it -> it.id.equals(id))
      .findFirst()
      .orElse(null);

    if (post != null) {
      return Mono.just(post);
    } else {
      return Mono.empty();
    }
  }

}

File: templates/kotlin-webmvc/subsystem/post/component/src/main/kotlin/{{PACKAGE_NAME}}/post/PostRepository.kt
package {{PACKAGE_NAME}}.post

import {{PACKAGE_NAME}}.post.model.Post
import org.springframework.stereotype.Repository

@Repository
internal class PostRepository {

  companion object {
    private val posts = listOf(
      Post(id = 1, userId = 1, title = "Lorem ipsum dolor sit amet.", body = "Curabitur ultrices lacus lorem, vitae tincidunt lorem convallis."),
      Post(id = 2, userId = 1, title = "Maecenas maximus sodales nibh vel.", body = "Pellentesque scelerisque tortor nec lorem volutpat, sed semper."),
      Post(id = 3, userId = 3, title = "Sed tempus, justo at viverra.", body = "Cras efficitur, nunc eu maximus congue, magna quam."),
      Post(id = 4, userId = 3, title = "Proin quis aliquam mauris. Proin.", body = "Donec id odio et mauris posuere auctor at."),
      Post(id = 5, userId = 3, title = "Nulla placerat condimentum diam sed.", body = "In et lectus a arcu efficitur viverra in."),
      Post(id = 6, userId = 3, title = "Etiam interdum facilisis urna, a.", body = "Phasellus euismod lectus eget libero viverra, sed rhoncus."),
      Post(id = 7, userId = 4, title = "Curabitur quis leo feugiat, faucibus.", body = "Phasellus rutrum pretium auctor. Donec tincidunt ligula eros."),
      Post(id = 8, userId = 5, title = "Praesent scelerisque gravida tortor a.", body = "Curabitur lobortis turpis at arcu facilisis, eu convallis."),
      Post(id = 9, userId = 5, title = "Vivamus nibh nulla, aliquet quis.", body = "Suspendisse consequat feugiat nisi vel sollicitudin. Phasellus nec."),
      Post(id = 10, userId = 5, title = "Aliquam elit erat, ornare nec.", body = "Phasellus finibus justo eget risus gravida, in venenatis."),
      Post(id = 11, userId = 6, title = "Duis vitae placerat nibh. Integer.", body = "Sed blandit nisl eget nisl gravida scelerisque. Pellentesque."),
      Post(id = 12, userId = 7, title = "Pellentesque eu tincidunt eros, sed.", body = "Nullam luctus dolor placerat lorem luctus, eu scelerisque."),
      Post(id = 13, userId = 7, title = "Nulla tincidunt cursus consequat. Aliquam.", body = "Praesent lobortis bibendum est eu volutpat. Maecenas lobortis."),
      Post(id = 14, userId = 7, title = "Etiam nec purus eget dui.", body = "Aenean interdum nec lacus eget elementum. Vivamus ultrices."),
      Post(id = 15, userId = 7, title = "In non est fringilla eros.", body = "Quisque malesuada posuere mauris, a pellentesque diam aliquet."),
      Post(id = 16, userId = 7, title = "Phasellus fermentum malesuada lectus nec.", body = "Nullam tortor magna, ullamcorper a elit a, aliquam.")
    )
  }

  fun findAll(): List<Post> = posts

  fun findByUserId(userId: Int): List<Post> = posts.filter { it.userId == userId }

  fun findById(id: Int): Post? = posts.find { it.id == id }

}
